### PR TITLE
Add method chaining style specification

### PR DESCRIPTION
**Pull request by vjik**
PR is inspired by https://github.com/yiisoft/docs/issues/155

Placing methods in a chain of methods on separate lines is more readable in most cases. In rare cases, this is not the case, so SHOULD.

### CODE REVIEW

File: spec.md [hunk 1]
```diff
@@ -649,6 +649,16 @@ $app->get('/hello/{name}', function ($name) use ($app) {
...
 ```
 
+Method chaining SHOULD be put on separate lines, where each subsequent line is indented once. When doing so, the first
+method MUST be on the next line.
+
+```php
+$someInstance
+    ->create()
+    ->prepare()
+    ->run();
```
Comment (xepozz) [inline spec.md:659]: Placing `;` on a separated line is handy as well.

Comment (xepozz) [reply spec.md:659]: It's just a little handy to comment/move/remove lines when you have `;` not at the end of the chain, but I think it's unnecessary to have it as a recommendation :)

Comment (andrewprofile) [reply spec.md:659]: in my opinion it is best to admit both versions :)

**Review by samdark — APPROVED** [review]

**Comment by KorvinSzanto** [comment]
I think we should change the "SHOULD" to be "MAY" and I agree that we should allow the parenthesis to be placed on the next line but maybe we can deal with that addition separately.

File: spec.md [hunk 2]
```diff
@@ -649,6 +649,16 @@ $app->get('/hello/{name}', function ($name) use ($app) {
 });
 ```
 
+Method chaining SHOULD be put on separate lines, where each subsequent line is indented once. When doing so, the first
```
Comment (samdark) [inline spec.md:652]:
```suggestion
Method chaining MAY be put on separate lines, where each subsequent line is indented once. When doing so, the first
```

**Comment by samdark** [comment]
Replaced SHOULD with MAY.

**Pull request merged**